src/components/OptionsSelector.js

```diff
@@ -87,7 +95,13 @@ class OptionsSelector extends Component {
     }
 
     componentDidMount() {
-        this.textInput.focus();
+        this.unsubscribeTransitionEnd = this.props.navigation.addListener('transitionEnd', () => {
+            this.textInput.focus();
+        });
+    }
+
+    componentWillUnmount() {
+        this.unsubscribeTransitionEnd();
```

Comment:
I could be wrong, but I thought this could be more useful in the `ScreenWrapper` and then we just pass a prop like `isScreenTransitioning`. Maybe it would be more complicated. Perhaps the same could be done with an HOC so we don't need to drill something like an `isScreenTransitioning` prop down.

Reply:
Of course, it depends on if there are other places where we'll need to do this. I think there already are..

Reply:
+1, this looks like something that'll be needed down the road